plugins/libplugin-pay.c

```diff
@@ -36,6 +36,7 @@ struct payment *payment_new(tal_t *ctx, struct command *cmd,
 		tal_arr_expand(&parent->children, p);
 		p->destination = parent->destination;
 		p->amount = parent->amount;
+		p->label = parent->label;
```

Comment:
Should it not also initialize this if there is no parent?  Consider the case where a new plugin does payments in the background without any commands or a concept of labels.

Reply:
Where would we get that label from? It's passed in from the JSON-RPC as an annotation, so we allow the caller to optionally add it to the root from where it'll get inherited, otherwise it defaults to `NULL`.